[10.x] Add option to ignore case on string helper method before and after

bmckay959 opened the pull request:
Added the option to ignore case on the `before` and `after` string helper. Set defaults to use the existing case sensitive functionality for backwards compatibility.

bmckay959 committed:
Initial commit - Add resume method to batched jobs as an option

bmckay959 committed:
add resume to command

bmckay959 committed:
added test for resume

bmckay959 committed:
add resume to BatchRepositoryFake

bmckay959 committed:
Update SupportStrTest.php

bmckay959 committed:
test and ignore case functionality for before and after string helper

bmckay959 committed:
Merge pull request #1 from bmckay959/bmckay959-ignore-case-str-helper-before-after

Bmckay959 ignore case str helper before after

bmckay959 commented on src/Illuminate/Support/Str.php:
```diff
@@ -83,8 +83,12 @@ public static function of($string)
      * @return string
      */
-    public static function after($subject, $search)
+    public static function after($subject, $search, $ignoreCase = false)
```
Default set for previous functionality

bmckay959 commented on src/Illuminate/Support/Str.php:
```diff
@@ -142,13 +146,17 @@ public static function transliterate($string, $unknown = '?', $strict = false)
      * @return string
      */
-    public static function before($subject, $search)
+    public static function before($subject, $search, $ignoreCase = false)
```
Default set for previous functionality

bmckay959 commented on tests/Support/SupportStrTest.php:
```diff
@@ -216,6 +216,25 @@ public function testStrBefore()
         $this->assertSame('han', Str::before('han0nah', '0'));
         $this->assertSame('han', Str::before('han0nah', 0));
         $this->assertSame('han', Str::before('han2nah', 2));
+        $this->assertSame('hannah', Str::before('hannah', 'NAH'));
```
Added a case sensitive to the previous functionality to show that it does not match

bmckay959 commented on tests/Support/SupportStrTest.php:
```diff
@@ -271,6 +290,25 @@ public function testStrAfter()
         $this->assertSame('nah', Str::after('han0nah', '0'));
         $this->assertSame('nah', Str::after('han0nah', 0));
         $this->assertSame('nah', Str::after('han2nah', 2));
+        $this->assertSame('hannah', Str::after('hannah', 'HAN'));
```
Added a case sensitive test to the previous functionality to show that it does not match

bmckay959 committed:
styling

bmckay959 committed:
Update Batch.php

Accidental Commit

bmckay959 committed:
remove batch files

bmckay959 committed:
Merge pull request #2 from bmckay959/bmckay959-patch-1

Remove unintended commits

bmckay959 commented:
Sorry for the mess of commits. I pushed up unintended code and had to clean it up.

morloderex commented on src/Illuminate/Support/Str.php:
```diff
@@ -83,8 +83,12 @@ public static function of($string)
+    public static function after($subject, $search, $ignoreCase = false)
     {
+        if ($ignoreCase) {
+            return $search === '' ? $subject : array_reverse(preg_split("/$search/i", $subject, 2))[0];
```
@bmckay959 would it make sense to use preg_split here now that we are using regular expressions?

```suggestion
            $search = preg_qoute($search);
            return $search === '' ? $subject : array_reverse(preg_split("/$search/i", $subject, 2))[0];
```

bmckay959 replied:
I think you comment meant, "use preg_quote here..."?

If so, I think the answer is yes. I'm not super familiar with regex and didn't even know preg_quote existed, I'll try to build out a few test cases that need to have something escaped today and add it.

Thanks!

bmckay959 committed:
Update src/Illuminate/Support/Str.php

Per @morloderex's suggestion to better escape user given strings on preg_split

Co-authored-by: Michael Lundbøl <michael.lundboel@gmail.com>

bmckay959 committed:
add tests for preg_quote

bmckay959 committed:
preg_quote typo

bmckay959 committed:
Merge branch '10.x' into bm/preg_quote_tests

bmckay959 committed:
Merge pull request #3 from bmckay959/bm/preg_quote_tests

add tests for preg_quote

bmckay959 committed:
styling

taylorotwell commented:
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

bmckay959 commented:
@taylorotwell I agree with the sentiment behind not bloating the framework. I would ask you to reconsider on this with the following scenario and if you still disagree, I can work around it, but it's too small a change to make into a package. Either way, grateful for you and the team.

See example below:

```php
$string = "LOOK FOR THIS! Additional text here...";
$search = "Look for this!";

// Note: ::contains allows ignoreCase as 3rd parameter
if(Str::contains($string, $search, true)) {

    // Currently, this does not find it and returns "LOOK FOR THIS! Additional text here..."
    $text = Str::after($string, $search);

    // Updates would allow this to return " Additional text here..."
    $text = Str::after($string, $search, true);
}
```
